core/services/workflows/model.go

```diff
@@ -0,0 +1,86 @@
+package workflows
+
+import (
+	"fmt"
+	"sigs.k8s.io/yaml"
+)
+
+func ParseWorkflowModelFromString(s string) (WorkflowModel, error) {
+    var model WorkflowModel
+    err := yaml.Unmarshal([]byte(s), &model)
+    if err != nil {
+        // TODO: Use chainlink/core/logger to log the error
+        return WorkflowModel{}, fmt.Errorf("failed to parse workflow model: %v", err)
+    }
+    return model, nil
+}
+
+// WorkflowModel represents the entire workflow configuration,
+// including its triggers, actions, consensus mechanisms, and targets.
+type WorkflowModel struct {
+    // Triggers define the events that initiate the workflow.
+    Triggers []Trigger `json:"triggers,omitempty"`
+
+    // Actions represent the set of operations performed within the workflow.
+    Actions []Action `json:"actions,omitempty"`
```

Comment:
@HenryNguyen5 I don't think we should have separate structs for Action, Consensus etc. They are intentionally supposed to be the same so I would suggest having one struct to cover all of those.

The one exception would be Trigger since IIRC there's a field that's missing (inputs I imagine).